user_tools/src/spark_rapids_tools/tools/qualification_stats_report.py

```diff
@@ -0,0 +1,124 @@
+                qual_output_dir, unsupported_operator_report_file)
+            self.unsupported_operators_df = pd.read_csv(rapids_unsupported_operators_file)
+
+            stages_report_file = self.ctxt.get_value('toolOutput', 'csv', 'stagesInformation', 'fileName')
+            rapids_stages_file = FSUtil.build_path(qual_output_dir, stages_report_file)
+            self.stages_df = pd.read_csv(rapids_stages_file)
+            self.logger.info('Reading CSV files completed.')
+        except Exception as e:  # pylint: disable=broad-except
+            self.logger.error('Error reading CSV files: %s', e)
+
+    def convert_durations(self):
+        try:
+            # Convert durations from milliseconds to seconds
+            self.unsupported_operators_df[['Stage Duration', 'App Duration']] /= 1000
+            self.stages_df[['Stage Task Duration', 'Unsupported Task Duration']] /= 1000
+        except Exception as e:  # pylint: disable=broad-except
+            self.logger.error('Error while converting durations to seconds: %s', e)
+
+    def merge_dataframes(self):
+        try:
+            self.logger.info('Merging dataframes to get stats...')
+            # Merge unsupported_operators_df with stages_df on App ID and Stage ID
+            merged_df = pd.merge(self.unsupported_operators_df, self.stages_df,
+                                 on=['App ID', 'Stage ID'])
+
+            agg_unsupported_df = (merged_df.groupby(['App ID', 'SQL ID', 'Unsupported Operator']).agg(
+                Count=('Unsupported Operator', 'size'),
+                Impacted_Stage_Duration=('Stage Duration', 'sum'),
+                App_Duration=('App Duration', 'first'),
+                Stage_Task_Duration=('Stage Task Duration', 'sum')
+            ).reset_index())
+
+            agg_unsupported_df['% of Stage Duration'] = (
+                    (agg_unsupported_df['Impacted_Stage_Duration'] /
+                     agg_unsupported_df['App_Duration']) * 100).round(3)
+
+            agg_unsupported_df['Supported'] = False
+            final_df = agg_unsupported_df.rename(columns={
+                'Unsupported Operator': 'Operator Name',
+                'Impacted_Stage_Duration': 'Impacted Stage duration(seconds)',
+                'Stage_Task_Duration': 'Stage Task Exec duration(seconds)'
+            })
+            self.result_df = final_df[self.output_columns.get('columns')].copy()
+            self.logger.info('Merging stats dataframes completed.')
+        except Exception as e:  # pylint: disable=broad-except
+            self.logger.error('Error merging stats dataframes: %s', e)
+
+    def write_results(self):
+        try:
+            self.logger.info('Writing stats results...')
+            result_output_dir = self.ctxt.get_output_folder()
+            outputfile_path = self.ctxt.get_value('local', 'output', 'files', 'statistics', 'name')
+            self.output_file = FSUtil.build_path(result_output_dir, outputfile_path)
```

Comment:
nit: Default output folder should probably have `stats_2024xxx` or `statistics_2024xxx` prefix
```
 INFO rapids.tools.qualification.stats: Results have been saved to /result/qual_20240729211337_2Cd791a7/qualification_statistics.csv
```